bigchaindb/common/transaction.py

```diff
@@ -925,16 +925,36 @@ def to_dict(self):
             'version': self.version,
         }
 
-        tx_no_signatures = Transaction._remove_signatures(tx)
-        tx_serialized = Transaction._to_str(tx_no_signatures)
-        tx['id'] = Transaction._to_hash(tx_serialized)
+        tx['id'] = Transaction._generate_tx_id(tx)
         if self.operation == Transaction.CREATE:
             # Duplicate asset into asset for consistency with TRANSFER
             # transactions
             tx['asset']['id'] = tx['id']
         return tx
 
     @staticmethod
+    def _generate_tx_id(tx):
+        """Generates an ID from a Transaction dictionary
+
+        Args:
+            tx (dict): The Transaction (as a dict) to get an ID from
+
+        Returns:
+            string: The Transaction's ID
```

Comment:
I think the return type needs to be an object, and if so in this case, it should be `str` instead of `string`.